### DISCUSSION TITLE
How to use contributes.markdown.previewScripts

### DISCUSSION THREAD

**Discussion by eai04191** [microsoft/vscode-discussions#2281] [Extension Development QnA] [answered]
I'm looking for how to use `contributes.markdown.previewScripts` because I want to add media to the markdown preview in the extension.

The documentation doesn't explain anything other than writing the path to the js file.

https://code.visualstudio.com/api/extension-guides/markdown-extension#adding-advanced-functionality-with-scripts

The documentation links to the Markdown Preview Mermaid Support extension, which works on its own, but doesn't work if you clone the source code and launch the extension.

<details>
<summary>Details</summary>

![](https://github.com/user-attachments/assets/1fda354d-e8aa-45e8-ac76-7972517e399e)

![](https://github.com/user-attachments/assets/d9cf7dc6-3573-4d1b-9ecb-e46a2e275a3c)

</details>

Here's a minimal duplicate that just calls alert(). Why doesn't this work? 
https://github.com/eai04191/previewscript

**Comment by robconery** [comment, marked as answer]
If I understand correctly, you want to add an image to the markdown chat output, correct? [Here's a snippet](https://github.com/microsoft/vscode-data-analysis-for-copilot/blob/main/src/tools.ts#L126) from the data extension which shows how to add a markdown image. If this isn't answering your question, would you mind adding more details?
